specs/AddHostObjectToScriptWithOrigins.md

```diff
@@ -0,0 +1,188 @@
+        // to this list.
+        int originsCount = 1;
+        LPCWSTR* originsList =
+            reinterpret_cast<LPCWSTR*>(CoTaskMemAlloc(sizeof(LPCWSTR) * originsCount));
+        originsList[0] = L"https://appassets.example";
+        // Add host object to a script for iframe access
+        CHECK_FAILURE(
+            webviewFrame->AddHostObjectToScriptWithOrigins(
+                L"sample", &remoteObjectAsVariant, originsList, originsCount));
+
+        remoteObjectAsVariant.pdispVal->Release();
+        CoTaskMemFree(originsList);
+
+        return S_OK;
+    }).Get(), &m_frameCreatedToken));
+}
+```
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+## IDL
+```c#
+/// WebView2Frame provides direct access to the iframes information and use.
+[uuid(18ae830b-8c83-459f-8a8c-95a1022af774), object, pointer_default(unique)]
+interface ICoreWebView2Frame : IUnknown {
+  /// The name of the iframe from the iframe html tag declaring it.
+  /// Calling this method fails if it is called after the iframe is deleted.
+  [propget] HRESULT Name([ out, retval ] LPWSTR * name);
+  /// Called when the iframe changes its window.name property
+  HRESULT add_NameChanged(
+      [in] ICoreWebView2FrameNameChangedEventHandler *eventHandler,
+      [out] EventRegistrationToken * token);
+  /// Remove an event handler previously added with add_NameChanged.
+  HRESULT remove_NameChanged([in] EventRegistrationToken token);
+
+  /// Add the provided host object to script running in the WebView2 with the
+  /// specified name for the list of the specified origins. The host object
+  /// will be accessible for this iframe only if the iframe's origin during
+  /// access matches one of the origins which are passed. Non-ASCII origins
+  /// can be passed in Unicode or Punycode format. The IDN origins will be
+  /// converted to Unicode and normalized before comparison. If the iframe is
+  /// declared with no src attribute its origin is considered the same as
+  /// the origin of the parent document.
+  /// List of origins will be treated as following:
+  /// 1. empty list - call will fail and no host object will be added for the
+  /// iframe;
+  /// 2. list with origins - during access to host object from iframe the
+  /// origin will be checked that it belongs to this list;
+  /// 3. list with "*" element - host object will be available for iframe for
+  /// all origins. We suggest not to use this feature without understanding
+  /// security implications of giving access to host object from from iframes
+  /// with unknown origins.
+  /// Calling this method fails if it is called after the iframe is deleted.
+  /// \snippet ScenarioAddHostObject.cpp AddHostObjectToScriptWithOrigins
+  /// For more information about host objects navigate to [AddHostObjectToScript]
+  HRESULT AddHostObjectToScriptWithOrigins(
+      [in] LPCWSTR name,
+      [in] VARIANT *object,
+      [in] LPCWSTR *origins,
```

Comment:
Pls put count parameter before origins parameter. Also we can also use the size_is attribute:

```
[in] UINT32 originsCount,
[in, size_is(originsCount)] LPCWSTR* origins);
```